doc/neps/nep-0054-simd-cpp-highway.rst

```diff
@@ -0,0 +1,480 @@
+
+
+Usage and Impact
+----------------
+
+N/A - there will be no significant user-visible changes.
+
+
+Backward compatibility
+----------------------
+
+There will be no changes in user-facing Python or C APIs: all the methods to
+control compilation and runtime CPU feature selection should remain, although
+there may be some changes due to moving to Meson and C++ without regards to the
+Highway/Universal Intrinsics choice.
+
+On Windows, MSVC may have to be avoided, as a result of Highway's use of
+pragmas which are less well supported by MSVC. This means that we likely have
+to build our wheels with clang-cl or Mingw-w64. Both of those should work - we
+merged clang-cl support a while back (see `gh-20866`_), and SciPy builds with
+Mingw-w64. It may however impact other redistributors or end users who build
+from source on Windows.
+
+The potential impact on licensing because of Highway's use of the Apache 2.0
+license needs discussing before Highway can be accepted. Or Highway would have
+to have to dual-license under Apache 2.0 and an MIT/BSD type license.
+
+
+High-level considerations
+-------------------------
+
+.. note::
+
+   Currently this section attempts to cover each topic separately, and
+   comparing the future use of a NumPy-specific C++ implementation vs. use of
+   Google Highway with our own numerical routines on top of that. It does not
+   (yet) assume a decision or proposed decision is made. Hence this NEP is not
+   "this is proposed" with another option in the Alternatives section, but
+   rather a side-by-side comparison.
+
+    
+Development effort and long-term maintainability
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Moving to Highway is likely to be a significant development effort.
+Longer-term, this will hopefully be offset by Highway itself having more
+maintainer bandwidth to deal with ongoing issues in compiler support and adding
+new platforms. It should increase the bus factor of SIMD support in NumPy to
+some extent. That said - the Highway repository also seems to show that its bus
+factor is low - it was 1 until the start of 2023, and now it's perhaps 2.
+
+Highway being used by other projects, like Chromium and `JPEG XL`_ (see
+`this more complete list <https://google.github.io/highway/en/master/README.html#examples>`__
+in the Highway documentation), does imply that there is likely to be a benefit
+of a wider range of testing and bug reporting/fixing.
+
+One concern is that new instructions may have to be added, and that that is
+often best done as part of the process of developing the numerical kernel that
+needs the instruction. This will be a little more clumsy if the instruction
+lives in Highway which is a git submodule inside the NumPy repo - there will be
+a need to implement a temporary/generic version first, and then update the
+submodule after upstreaming the new intrinsic.
+https://numpy.org/doc/1.25/reference/simd/
+
+Documentation-wise, Highway would be a clear win. NumPy's
```

Comment:
gh-21057 provides a rich reference too auto generated by doxygen, check [build artifact]( https://output.circle-artifacts.com/output/job/6a58c19b-2137-4b00-80a7-50e532c4d7b3/artifacts/0/doc/build/html/reference/simd/intrinsics/index.html)

Reply:
I agree that the documentation of the work-in-progress C++ functions is nicely structured and thorough. There are additional topics that can be interesting, e.g. an FAQ, implementation details, design rationale. If we are pooling our efforts, maybe we can combine what's best from both sets of documentation :)